src/devices/vdml_ext_adi.c

```diff
@@ -457,44 +458,46 @@ ext_adi_led_t ext_adi_led_init(uint8_t smart_port, uint8_t adi_port) {
 	transform_adi_port(adi_port);
 	claim_port_i(smart_port - 1, E_DEVICE_ADI);
 	vexDeviceAdiPortConfigSet(device->device_info, adi_port, (V5_AdiPortConfiguration)E_ADI_DIGITAL_OUT); 
-	return_port(smart_port - 1, merge_adi_ports(smart_port, adi_port + 1));
+	return_port(smart_port - 1, merge_adi_ports(smart_port - 1, adi_port+1));
 }
 
 int32_t ext_adi_led_set(ext_adi_led_t led, uint32_t* buffer, uint32_t buffer_length) {
 	uint8_t smart_port, adi_port;
 	get_ports(led, smart_port, adi_port);
-	claim_port_i(smart_port - 1, E_DEVICE_ADI);
 	transform_adi_port(adi_port);
-	validate_type(device, adi_port, smart_port - 1, E_ADI_DIGITAL_OUT);//subtracted 1 from smart_port
+	claim_port_i(smart_port, E_DEVICE_ADI);
+	validate_type(device, adi_port, smart_port, E_ADI_DIGITAL_OUT);
 	if (buffer_length > MAX_LED) {
 		buffer_length = MAX_LED;
 	}
-	else if (buffer == NULL || buffer_length < 1 || buffer_length)
+	else if (buffer == NULL || buffer_length < 1)
 	{
 		errno = EINVAL;
 		return PROS_ERR;
 	}
 	uint32_t rtv = (uint32_t)vexDeviceAdiAddrLedSet(device->device_info, adi_port, buffer, 0, buffer_length, 0);
-	return_port(smart_port - 1, rtv); 
+	return_port(smart_port, rtv); 
 }
 
 int32_t ext_adi_led_set_pixel(ext_adi_led_t led, uint32_t* buffer, uint32_t buffer_length, uint32_t color, uint32_t pixel_position) {
 	uint8_t smart_port, adi_port;
 	get_ports(led, smart_port, adi_port);
-	claim_port_i(smart_port - 1, E_DEVICE_ADI); 
+	claim_port_i(smart_port, E_DEVICE_ADI); 
 	transform_adi_port(adi_port);
-	validate_type(device, adi_port, smart_port - 1, E_ADI_DIGITAL_OUT);
+	validate_type(device, adi_port, smart_port, E_ADI_DIGITAL_OUT);
 	if(buffer == NULL || pixel_position < 0 || buffer_length >= MAX_LED || buffer_length < 1 || pixel_position > buffer_length - 1) {
 		errno = EINVAL;
-		return_port(smart_port - 1, PROS_ERR);
+		return_port(smart_port, PROS_ERR);
 	}
 	buffer[pixel_position] = color;
 	uint32_t rtv = (uint32_t)vexDeviceAdiAddrLedSet(device->device_info, adi_port, buffer, 0, buffer_length, 0);
 	return_port(smart_port - 1, rtv); 
 }
 
 int32_t ext_adi_led_set_all(ext_adi_led_t led, uint32_t* buffer, uint32_t buffer_length, uint32_t color) {
-	memset(buffer, color, buffer_length * sizeof(uint32_t));
+	for(int i = 0;i<buffer_length;i++){
+      buffer[i] = color;
+  	}
```

Comment:
the idea behind doing a memset instead of a for loop is because memset is much more atomic than a for loop... are we sure memset does not work here and what was the issue with it?

Reply:
I tried playing with it quite a bit and it didn't work. 

fwiw, jpearman does it this same way with for loops in the vexcode implementation of addrled.